presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -554,9 +563,23 @@ public PlanOptimizers(
                                 new AddNotNullFiltersToJoinNode(metadata.getFunctionAndTypeManager()))),
                 inlineProjections,
                 simplifyRowExpressionOptimizer, // Re-run the SimplifyExpressions to simplify any recomposed expressions from other optimizations
+                inlineProjections,
                 projectionPushDown,
                 new PayloadJoinOptimizer(metadata),
                 new UnaliasSymbolReferences(metadata.getFunctionAndTypeManager()), // Run again because predicate pushdown and projection pushdown might add more projections
+                new ReplaceConstantVariableReferencesWithConstants(metadata.getFunctionAndTypeManager()),
```

Comment:
Can you add a comment on why we need to repeat these here ?

Reply:
It's because we may have some new constant variables after the first run, added the comment.